src/targets/maven.ts

```diff
@@ -393,7 +448,10 @@ export class MavenTarget extends BaseTarget {
         );
       }
     }
-
+    // If klib target file name exists, use that instead of .jar
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}.klib`))) {
+      return `${moduleName}.klib`;
+    }
```

Comment:
If the file isn't accessible (e.g. it doesn't exist), the `.jar` filename will be returned. I don't know if the `.jar` file exists for kotlin, or what the consequences are of uploading it instead of the `.klib`. If a required file for a release doesn't exist, we should fail the release.

Another thing to consider is how easy it'd be to debug a failing release. A message like "hey, I can't find xxx file" is more helpful than "hey, I'm uploading the files and can't read yyy file, which isn't related to the release at all".

An alternative approach is to update the target's config to keep track of what we are releasing (android? kotlin?). If we know that, we could return directly the `.klib` filename, and let the upload fail if it can't read the file (which automatically produces a more helpful failure message). This is what currently happens with the `.jar` file (no disk operations happen in this method).

Reply:
I guess what I can do is create a target config for `kotlin-multiplatform` or something like that to keep the `uploadPomDistribution` separated and only add those files if it's a `kotlin-multiplatform` artifact. that way it doesn't interfere with the way it has been before I added the changes